icu4c/source/common/uassert.h

```diff
@@ -31,6 +31,8 @@
 #if U_DEBUG
 #   include <assert.h>
 #   define U_ASSERT(exp) assert(exp)
+#elif __cplusplus
```

Comment:
Could this be something like:
`#elif (U_CPLUSPLUS_VERSION > 0)`
to avoid the warning maybe?

Reply:
Thanks, good idea!  Let's see if this works...

```
#elif U_CPLUSPLUS_VERSION
```